.eslintrc.js

```diff
@@ -42,12 +42,20 @@ module.exports = {
     'no-fallthrough': 1,
     'no-lone-blocks': 1,
     'no-delete-var': 1,
-    'no-shadow': 1,
+    'no-shadow': 'off',
+    '@typescript-eslint/no-shadow': 'warn',
     'no-shadow-restricted-names': 1,
     'no-undef': 2,
     'no-undef-init': 1,
     'no-use-before-define': 0,
-    'no-unused-vars': [1, { vars: 'all', args: 'none' }],
+    'no-unused-vars': 'off', // This must be disabled for the typescript rule below to work
+    '@typescript-eslint/no-unused-vars': [
+      'warn',
+      {
+        vars: 'all',
+        args: 'none', // We should change to args: after-used
+      },
+    ],
```

Comment:
Yes it will. This is in the rule [documentation instructions](https://typescript-eslint.io/rules/no-unused-vars/#how-to-use).

The only change I left documented for later is the `args: after-used` option, that would require too many refactorings and possibly breaking changes.